packages/react-instantsearch-core/src/core/indexUtils.js

```diff
@@ -69,24 +69,27 @@ export function refineValue(
 function refineMultiIndex(searchState, nextRefinement, context, resetPage) {
   const page = resetPage ? { page: 1 } : undefined;
   const indexId = getIndexId(context);
-  const state = has(searchState, `indices.${indexId}`)
-    ? {
-        ...searchState.indices,
-        [indexId]: {
-          ...searchState.indices[indexId],
-          ...nextRefinement,
-          ...page,
-        },
-      }
-    : {
-        ...searchState.indices,
-        ...{
+  const state =
+    searchState &&
```

Comment:
We can drop this first condition because in both close of the ternary we use `searchState.indices`.